File: OOP/OOP/source_files/groupLeader.cpp
#include <iostream>
#include "../header_files/groupLeader.h"

using namespace std;


GroupLeader::GroupLeader() : Student() {};


GroupLeader::GroupLeader(std::istream &is) : Student(is) {
    cout << "email: ";
    getline(cin >> std::ws, this->email);
    cout << "номер телефона: ";
    getline(cin >> std::ws, this->phoneNumber);
}


void GroupLeader::writeToConsole() const {
    Student::writeToConsole();
    cout << "email: " << this->email << endl;
    cout << "номер телефона: " << this->phoneNumber << endl;
}

File: SHARP/StudentsManager/GroupLeader.cs
using System;
using System.Text.Json.Serialization;


namespace StudentManager
{
    public class GroupLeader : Student
    {
        public string Email { get; set; } = string.Empty;
        public string PhoneNumber { get; set; } = string.Empty;

        [JsonIgnore]
        public override string ClassName => "староста";
        [JsonIgnore]
        public new static string StaticClassName => "староста";

        public GroupLeader() : base() { }

        public GroupLeader(bool fromConsole) : base(fromConsole)
        {
            if (fromConsole)
            {
                Console.Write("Email: ");
                this.Email = Console.ReadLine() ?? string.Empty;
                
                Console.Write("Номер телефона: ");
                this.PhoneNumber = Console.ReadLine() ?? string.Empty;
            }
        }

        public override void WriteToConsole()
        {
            base.WriteToConsole();

            Console.WriteLine($"Email: {Email}");
            Console.WriteLine($"Номер телефона: {PhoneNumber}");
        }
    }
}
